faunadb-java-dsl/src/main/java/com/faunadb/client/types/Decoder.java

```diff
@@ -0,0 +1,211 @@
+package com.faunadb.client.types;
+
+import com.faunadb.client.errors.FaunaException;
+import com.faunadb.client.types.Types.CollectionType;
+import com.faunadb.client.types.Types.MapType;
+import com.faunadb.client.types.Types.SimpleType;
+import com.faunadb.client.types.Value.BytesV;
+import com.faunadb.client.types.Value.NullV;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Type;
+import java.util.*;
+
+import static com.google.common.base.Defaults.defaultValue;
+import static java.lang.String.format;
+import static java.lang.reflect.Modifier.isAbstract;
+
+/**
+ * FaunaDB {@link Value} to object decoder.
+ */
+@SuppressWarnings("unchecked")
+public final class Decoder {
+  private Decoder() {
+  }
+
+  /**
+   * Decode a FaunaDB {@link Value} to a specified type.
+   * <p>
+   * <p>This method is useful if you need to decode
+   * collections types.</p>
```

Comment:
i think you can drop 'types' and just say 'This method is useful if you need to decode collections.'